### PR TITLE
feat(server): meta schema fields

### PR DESCRIPTION
**Pull request by mimoham24**

### CODE REVIEW

File: server/internal/usecase/interfaces/model.go [hunk 1]
```diff
@@ -39,6 +38,7 @@ type Model interface {
 	FindByProject(context.Context, id.ProjectID, *usecasex.Pagination, *usecase.Operator) (model.List, *usecasex.PageInfo, error)
 	FindByKey(context.Context, id.ProjectID, string, *usecase.Operator) (*model.Model, error)
 	FindByIDOrKey(context.Context, id.ProjectID, model.IDOrKey, *usecase.Operator) (*model.Model, error)
+	CreateMetadata(context.Context, id.ModelID, *usecase.Operator) (*model.Model, error)
```
Comment (rot1024) [inline server/internal/usecase/interfaces/model.go:41]: is it used?

File: server/internal/usecase/interactor/item.go [hunk 2]
```diff
@@ -304,6 +305,19 @@ func (i Item) Update(ctx context.Context, param interfaces.UpdateItemParam, oper
 		oldFields := itv.Fields()
 
 		itv.UpdateFields(fields)
+
+		if operator.AcOperator.User != nil {
+			itv.SetUpdatedByUser(*operator.AcOperator.User)
+		}
+
+		if operator.Integration != nil {
```
Comment (rot1024) [inline server/internal/usecase/interactor/item.go:313]:
```suggestion
		} else if operator.Integration != nil {
```

File: server/internal/usecase/interactor/item.go [hunk 3]
```diff
@@ -208,6 +208,7 @@ func (i Item) Create(ctx context.Context, param interfaces.CreateItemParam, oper
 			Schema(s.ID()).
 			Project(s.Project()).
 			Model(m.ID()).
+			MetadataItem(param.MetadataID).
```
Comment (rot1024) [inline server/internal/usecase/interactor/item.go:211]: MetadaID seems not validated

File: server/internal/usecase/interactor/model.go [hunk 4]
```diff
@@ -69,6 +69,10 @@ func (i Model) Create(ctx context.Context, param interfaces.CreateModelParam, op
 			if err != nil {
 				return nil, err
 			}
+			meta, err := schema.New().NewID().Workspace(p.Workspace()).Project(p.ID()).Build()
```
Comment (rot1024) [inline server/internal/usecase/interactor/model.go:72]: Do you always create metadata? Not all models use metadata, so it is fine to create it on a delayed basis (when fields are added to the metadata for the first time).

File: server/internal/adapter/gql/resolver_mutation_item.go [hunk 5]
```diff
@@ -23,10 +23,14 @@ func (r *mutationResolver) CreateItem(ctx context.Context, input gqlmodel.Create
 	if err != nil {
 		return nil, err
 	}
+	if err != nil {
+		return nil, err
+	}
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_item.go:28]:
```suggestion

```

File: server/internal/adapter/gql/resolver_mutation_field.go [hunk 6]
```diff
@@ -68,18 +67,25 @@ func (r *mutationResolver) UpdateField(ctx context.Context, input gqlmodel.Updat
...
+		s, err = usecases(ctx).Schema.FindByID(ctx, *m.Metadata(), getOperator(ctx))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		s, err = usecases(ctx).Schema.FindByID(ctx, m.Schema(), getOperator(ctx))
+		if err != nil {
+			return nil, err
+		}
 	}
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_field.go:89]: These logic should be moved to the usecase layer. This is the adapter layer, so it should be thin.

File: server/internal/usecase/interactor/model.go (hunk 4 above)
Comment (rot1024) [reply server/internal/usecase/interactor/model.go]: Is it possible to add fields to the metadata without the metadata existing in the schema? Does an implementation exist?

Comment (mimoham24) [reply server/internal/usecase/interactor/model.go]: metadata is created on field creation for the first time

File: server/i18n/ja.yml [hunk 7]
```diff
@@ -77,4 +77,5 @@ unsupported operation: サポートされていない処理です。
 uuid is required: UUIDは必須です。
 value is required: 値は必須です。
 workspace id is required: ワークスペースIDは必須です。
-metadata schema not found: ""
\ No newline at end of file
+metadata schema not found: ""
+metadata item and schema mismatch: ""
```
Comment (rot1024) [inline server/i18n/ja.yml:81]:
```suggestion
metadata schema not found: メタデータのスキーマが見つかりません。
metadata item and schema mismatch: メタデータのアイテムのスキーマが正しくありません。
```

File: server/internal/adapter/gql/resolver_mutation_field.go [hunk 8]
```diff
@@ -35,7 +43,7 @@ func (r *mutationResolver) CreateField(ctx context.Context, input gqlmodel.Creat
...
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_mutation_field.go:43]: line 22~43 should be done in the usecase layer. CreateField usecase should accept modelID and metadata bool. The adapter layer should be thin as possible.

Comment (mimoham24) [reply server/internal/adapter/gql/resolver_mutation_field.go:43]: do you mean line 22~39?  lines 40~43 are using a gqlmodel convertor and we are following the same logic everywhere, If so we need to move this logic from the update function as well

File: server/internal/usecase/interfaces/model.go [hunk 9]
```diff
@@ -20,6 +20,11 @@ type CreateModelParam struct {
 	Key         *string
 	Public      *bool
 }
+type FindOrCreateSchemaParam struct {
+	ModelId  id.ModelID
+	Metadata *bool
+	Create   bool
```
Comment (rot1024) [inline server/internal/usecase/interfaces/model.go:26]:
```suggestion
	ModelID  id.ModelID
	// boolean that identify if it is a metdata
	Metadata *bool
	// boolean to identify if we want to create a metadata schema or just return an error if metadata schema is nil
	Create   bool
```

File: server/internal/usecase/interfaces/model.go [hunk 10]
```diff
@@ -20,6 +20,11 @@ type CreateModelParam struct {
 	Key         *string
 	Public      *bool
 }
+type FindOrCreateSchemaParam struct {
```
Comment (rot1024) [inline server/internal/usecase/interfaces/model.go:23]:
```suggestion

type FindOrCreateSchemaParam struct {
```

insert line break between struct

File: server/internal/usecase/interfaces/model.go [hunk 11]
```diff
@@ -20,6 +20,11 @@ type CreateModelParam struct {
...
+	ModelId  id.ModelID
+	Metadata *bool
+	Create   bool
+}
 
 type UpdateModelParam struct {
 	ModelId     id.ModelID
```
Comment (rot1024) [inline server/internal/usecase/interfaces/model.go:30]:
```suggestion
	ModelID     id.ModelID
```

**Review by rot1024 — APPROVED** [review]

**Pull request merged**